--- sqlplusint/type_info1.hh.orig	Thu Nov 27 18:39:41 2003
+++ sqlplusint/type_info1.hh	Thu Nov 27 18:40:43 2003
@@ -26,7 +26,8 @@
   const std::type_info     *_c_type;
   const unsigned char  _base_type;
   const bool           _default;
-  mysql_ti_sql_type_info (const mysql_ti_sql_type_info &b);           // can't do
+  mysql_ti_sql_type_info (const mysql_ti_sql_type_info &b)
+    : _sql_name (b._sql_name), _c_type (b._c_type), _base_type (b._base_type), _default (b._default) {}
   mysql_ti_sql_type_info& operator=(const mysql_ti_sql_type_info &b); //  "    "
   mysql_ti_sql_type_info () {} 
   // OEP - didn't init _base_type and _default mysql_ti_sql_type_info () {} 
