### PATCH shift-osx
--- config/linux.cf.orig	Mon Oct 20 23:09:47 2003
+++ config/linux.cf	Mon Oct 20 23:10:36 2003
@@ -9,12 +9,11 @@
 
 #define linux 1
 #define HasRanlib	YES
-CC += -fPIC -D_LARGEFILE64_SOURCE
-MTCCFLAGS = -pthread -DCTHREAD_LINUX -D_THREAD_SAFE -D_REENTRANT
+MTCCFLAGS =
 MTLDLIBS =
-MTLDFLAGS = -pthread
-SHLIBLDFLAGS = -shared
-SHLIBREQLIBS = -lc -lpthread
+MTLDFLAGS =
+SHLIBLDFLAGS = -dynamic
+SHLIBREQLIBS = -lSystem
 #if UseAdns
 ADNSLIB = -ladns
 #endif
--- ./commands/Imakefile	Thu Nov 11 15:36:53 1999
+++ ../.build/macosx26_gcc31/shift/2003/commands/Imakefile	Wed Oct  8 13:33:36 2003
@@ -22,7 +22,7 @@
 #endif
 #else
 #if defined(SOLARIS) || defined(__Lynx__) || defined(linux)
-LDLIBS = -L../common -lcommon $(ADNSLIB) -lnsl
+LDLIBS = -L../common -lcommon
 #else
 #if _WIN32
 LDLIBS = ..\common\common.lib advapi32.lib
--- ./common/Imakefile	Thu Nov 11 15:36:56 1999
+++ ../.build/macosx26_gcc31/shift/2003/common/Imakefile	Wed Oct  8 13:48:35 2003
@@ -91,6 +66,7 @@
 
 .PRECIOUS:	LibraryTargetName(common)
 
+SYSDEP_OBJS = cuserid.o
 COMMONLIB_OBJS = $(SYSDEP_OBJS) \
 		 compat.Osuf getacct.Osuf getacctent.Osuf getconfent.Osuf \
 		 getfilep.Osuf getifnam.Osuf getuser.Osuf log.Osuf readc.Osuf \
--- ./common/sendscsicmd.c	Thu Nov 11 15:37:15 1999
+++ ../.build/macosx26_gcc31/shift/2003/common/sendscsicmd.c	Wed Oct  8 13:44:15 2003
@@ -51,7 +51,9 @@
 #endif
 #if defined(linux)
 #include <stdlib.h>
+#ifndef __APPLE__
 #include SCSIINC
+#endif
 #include <sys/stat.h>
 #endif
 #include "scsictl.h"
@@ -65,7 +67,7 @@
 #else
 #define USRMSG(fmt,p,f,msg) {}
 #endif
-#if !defined(linux)
+#if !defined(linux) && !defined(__APPLE__)
 extern char *sys_errlist[];
 #endif
 static char nosensekey[] = "no sense key available";
@@ -543,7 +545,7 @@
 	}
 	return (buflen - ccb.cam_resid);
 #else
-#if defined(linux)
+#if defined(linux) && !defined(__APPLE__)
 	int fd;
 	FILE *fopen();
 	int n;
--- tape/usrlbl.c.orig	Mon Oct 20 17:08:11 2003
+++ tape/usrlbl.c	Mon Oct 20 17:09:21 2003
@@ -139,7 +139,7 @@
 	memcpy (hdr2, labelid, 3);
 
 	if (dlip->lblcode == SL) asc2ebc (hdr1, 80);
-#if defined(hpux) || defined(linux)
+#if (defined(hpux) || defined(linux)) && !defined(__APPLE__)
 	/* Please note that under Linux, this call is not necessary here
 	because the previous writing of filemark has already cleared the
 	EOT condition */
@@ -163,7 +163,7 @@
 	if (labelid[2] == 'V')	/* must clear EOT condition */
 		c = ioctl (tapefd, MTIOCGET, &mt_info);
 #endif
-#if defined(linux)
+#if defined(linux) && !defined(__APPLE__)
 	/* must clear EOT condition */
 	c = ioctl (tapefd, MTIOCGET, &mt_info);
 #endif
@@ -186,7 +186,7 @@
 		if (labelid[2] == 'V')	/* must clear EOT condition */
 			c = ioctl (tapefd, MTIOCGET, &mt_info);
 #endif
-#if defined(linux)
+#if defined(linux) && !defined(__APPLE__)
 		/* must clear EOT condition */
 		c = ioctl (tapefd, MTIOCGET, &mt_info);
 #endif
--- ns/Cns_readdir.c.orig	Mon Oct 20 17:43:14 2003
+++ ns/Cns_readdir.c	Mon Oct 20 17:43:22 2003
@@ -10,10 +10,10 @@
 /*	Cns_readdir - read a directory entry */
 
 #include <errno.h>
+#include <sys/types.h>
 #include <dirent.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #else
@@ -105,7 +105,9 @@
 		while (nbentries--) {
 			dp->d_ino = 0;
 #if defined(linux) || defined(sgi) || defined(SOLARIS)
+#ifndef __APPLE__
 			dp->d_off = 0;
+#endif
 #endif
 #if defined(_AIX)
 			dp->d_offset = 0;
--- ns/Cns_readdirc.c.orig	Mon Oct 20 17:58:12 2003
+++ ns/Cns_readdirc.c	Mon Oct 20 17:58:28 2003
@@ -10,10 +10,10 @@
 /*	Cns_readdirc - read a directory entry including comments */
 
 #include <errno.h>
+#include <sys/types.h>
 #include <dirent.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #else
--- msg/term.c.orig	Mon Oct 20 17:15:40 2003
+++ msg/term.c	Thu Aug 31 07:53:20 2000
@@ -21,7 +21,7 @@
 #if !defined(linux)
 #include <sys/termio.h>
 #else
-#include <termio.h>
+#include <termios.h>
 #endif
 #endif /* _WIN32 */
 #include <sys/types.h>
--- tape/smcsubr.c.orig	Mon Oct 20 17:14:05 2003
+++ tape/smcsubr.c	Mon Oct 20 17:11:22 2003
@@ -17,7 +17,7 @@
 #include "serrno.h"
 #include "smc.h"
 #define	RBT_XTRA_PROC 10
-#if !defined(linux)
+#if !defined(linux) && !defined(__APPLE__)
 extern char *sys_errlist[];
 #endif
 static struct smc_status smc_status;
--- rtcopy/Imakefile.orig	Mon Oct 20 18:17:20 2003
+++ rtcopy/Imakefile	Mon Oct 20 18:17:59 2003
@@ -235,19 +235,19 @@
 			$(CP) rtcp_CheckReq.c rtcpc_CheckReq.c
 
 rtcpc_SendRecv.Osuf:	rtcpc_SendRecv.c ../h/rtcp.h ../h/rtcp_constants.h ../h/rtcp_api.h
-	$(CC) -c $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) rtcpc_SendRecv.c
+	$(CC) -c $(CPPFLAGS) $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) rtcpc_SendRecv.c
 rtcpc_log.Osuf:		rtcpc_log.c ../h/rtcp.h ../h/rtcp_constants.h ../h/rtcp_api.h
-	$(CC) -c $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) rtcpc_log.c
+	$(CC) -c $(CPPFLAGS) $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) rtcpc_log.c
 rtcpc_CallTMS.Osuf:	rtcpc_CallTMS.c ../h/rtcp.h ../h/rtcp_constants.h ../h/rtcp_api.h
-	$(CC) -c $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) $(TMSFLG) rtcpc_CallTMS.c
+	$(CC) -c $(CPPFLAGS) $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) $(TMSFLG) rtcpc_CallTMS.c
 rtcpc_CallVMGR.Osuf:	rtcpc_CallVMGR.c ../h/rtcp.h ../h/rtcp_constants.h ../h/rtcp_api.h
-	$(CC) -c $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) $(VMGRFLG) rtcpc_CallVMGR.c
+	$(CC) -c $(CPPFLAGS) $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) $(VMGRFLG) rtcpc_CallVMGR.c
 rtcpc_CheckReq.Osuf:	rtcpc_CheckReq.c ../h/rtcp.h ../h/rtcp_constants.h ../h/rtcp_api.h
-	$(CC) -c $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) rtcpc_CheckReq.c
+	$(CC) -c $(CPPFLAGS) $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) rtcpc_CheckReq.c
 tpread.Osuf:		tpread.c ../h/rtcp.h ../h/rtcp_constants.h ../h/rtcp_api.h
-	$(CC) -c $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) tpread.c
+	$(CC) -c $(CPPFLAGS) $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) tpread.c
 dumptape.Osuf:		dumptape.c ../h/rtcp.h ../h/rtcp_constants.h ../h/rtcp_api.h
-	$(CC) -c $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) dumptape.c
+	$(CC) -c $(CPPFLAGS) $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) dumptape.c
 #if BuildRtstat
 rtstat.Osuf:		rtstat.c ../h/vdqm.h ../h/osdep.h ../h/net.h
 	$(CC) -c $(CFLAGS) rtstat.c
--- ./common/serror.c	Thu Nov 11 15:37:16 1999
+++ ../.build/macosx26_gcc31/shift/2003/common/serror.c	Wed Oct  8 13:33:36 2003
@@ -59,6 +61,7 @@
 #define strerror DECC$STRERROR
 #endif /* vms && __alpha && COMPILE_NOPREFIX */
 
+int     serrno = 0;
 char    *sys_serrlist[SEMAXERR-SEBASEOFF+2]=
         {"Error 0",
          "Host not known",
--- rfio/preseek64.c.orig	Mon Oct 20 17:02:17 2003
+++ rfio/preseek64.c	Mon Oct 20 17:02:35 2003
@@ -18,6 +18,7 @@
  */
 #define RFIO_KERNEL     1  
 #ifdef linux
+#include <sys/types.h>
 #include <sys/uio.h>
 #endif
 #include "rfio.h"     
--- ./copytape/copytape.c	Thu Nov 11 15:37:40 1999
+++ ../.build/macosx26_gcc31/shift/2003/copytape/copytape.c	Wed Oct  8 13:33:36 2003
@@ -11,7 +11,6 @@
 #include <signal.h>
 #include <sys/types.h>
 #include <grp.h>
-#include <malloc.h>
 #include <pwd.h>
 #include <string.h>
 #if defined(_WIN32)
--- ./copytape/tpcopy.c	Thu Nov 11 15:37:46 1999
+++ ../.build/macosx26_gcc31/shift/2003/copytape/tpcopy.c	Wed Oct  8 13:33:36 2003
@@ -11,7 +11,6 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <fcntl.h>
-#include <malloc.h>
 #include <netinet/in.h>
 #include <signal.h>
 #include <sys/time.h>
--- ./examples/Imakefile	Thu Nov 11 15:37:53 1999
+++ ../.build/macosx26_gcc31/shift/2003/examples/Imakefile	Wed Oct  8 13:33:36 2003
@@ -35,7 +35,7 @@
 LDLIBS = -L../lib -lshift -lsun $(ULTRA)
 #else /* ! hpux && ! cray && ! sgi */
 #if SOLARIS
-LDLIBS = -L../lib -lshift -lsocket -lnsl $(ULTRA)
+LDLIBS = -L../lib -lshift -lsocket $(ULTRA)
 #else /* ! hpux && ! cray && ! sgi && ! SOLARIS */
 #if _WIN32
 LDLIBS = ..\lib\shift.lib wsock32.lib advapi32.lib
--- ./rfio/dump_log.c	Thu Nov 11 15:38:50 1999
+++ ../.build/macosx26_gcc31/shift/2003/rfio/dump_log.c	Wed Oct  8 13:33:36 2003
@@ -9,7 +9,6 @@
 /* dump_log.c : A command to dump a client log */
 
 #include <stdio.h>
-#include <malloc.h>
 #include <sys/types.h>
 #include <sys/time.h>
 #include <time.h>
--- ./rfio/logging.c	Thu Nov 11 15:39:00 1999
+++ ../.build/macosx26_gcc31/shift/2003/rfio/logging.c	Wed Oct  8 13:33:36 2003
@@ -18,7 +18,6 @@
 #include <time.h>
 #include <fcntl.h>
 #include <unistd.h>
-#include <malloc.h>
 #include <syslog.h>
 #include <sys/uio.h>
 #include "logging.h"
--- ./rfio/open.c	Thu Nov 11 15:39:06 1999
+++ ../.build/macosx26_gcc31/shift/2003/rfio/open.c	Wed Oct  8 13:33:36 2003
@@ -25,6 +25,6 @@
 #endif

-RFILE  *rfilefdt[MAXRFD];        /* File descriptors tables             */
+RFILE  *rfilefdt[MAXRFD] = {};        /* File descriptors tables             */
 
 /*
  * General purpose buffer.
--- ./rfio/preseek.c	Thu Nov 11 15:39:11 1999
+++ ../.build/macosx26_gcc31/shift/2003/rfio/preseek.c	Wed Oct  8 13:33:36 2003
@@ -14,6 +14,7 @@
  */
 #define RFIO_KERNEL     1  
 #ifdef linux
+#include <sys/types.h>
 #include <sys/uio.h>
 #endif
 #include "rfio.h"     
--- ./rfio/rfio_calls.c	Thu Nov 11 15:39:19 1999
+++ ../.build/macosx26_gcc31/shift/2003/rfio/rfio_calls.c	Wed Oct  8 13:33:36 2003
@@ -2053,6 +2055,9 @@
    sigvec(SIGCHLD, &sv, (struct sigvec *)0);
 #endif /* sun || ultrix || _AIX */
 #if defined(sgi) || defined (hpux) || defined (SOLARIS) || (defined(__osf__) && defined(__alpha)) || defined(linux)
+#ifndef SIGCLD
+# define SIGCLD SIGCHLD
+#endif
    signal(SIGCLD, SIG_DFL) ;
 #endif /* sgi || hpux || SOLARIS || alpha-osf || linux */
    p= rqstbuf ;
--- common/serror.c.orig	Sun Nov 16 16:04:08 2003
+++ common/serror.c	Sun Nov 16 16:03:48 2003
@@ -17,7 +17,7 @@
 #include <Cglobals.h>
 #include <string.h>
 
-#ifndef linux
+#if !defined (linux) && !defined(__APPLE__)
 extern int      sys_nerr;       /* number of system error messages      */
 #endif
 
--- rfio/rfio_serv.c.orig	Mon Oct 20 23:36:51 2003
+++ rfio/rfio_serv.c	Mon Oct 20 23:37:04 2003
@@ -438,7 +438,7 @@
 #if HPUX10
             setpgrp3();
 #else
-#if __Lynx__
+#if __Lynx__ || __APPLE__
             setpgrp(0, getpid());
 #else
             setpgrp();
--- ./rfio/rfstatfs.c	Thu Nov 11 15:39:29 1999
+++ ../.build/macosx26_gcc31/shift/2003/rfio/rfstatfs.c	Wed Oct  8 13:33:36 2003
@@ -24,9 +24,11 @@
 #include <sys/mount.h>
 #endif
 #if (defined( sun ) && !defined(SOLARIS)) || defined( hpux ) || defined(linux)
+#if !defined(__APPLE__)
 #include <sys/vfs.h>
 #endif
-#if defined(ultrix)
+#endif
+#if defined(ultrix) || defined(__APPLE__)
 #include <sys/types.h>
 #include <sys/param.h>
 #include <sys/mount.h>
--- ./rtcopy/rtstat.c	Thu Nov 11 15:39:50 1999
+++ ../.build/macosx26_gcc31/shift/2003/rtcopy/rtstat.c	Wed Oct  8 13:33:36 2003
@@ -10,7 +10,6 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <grp.h>
-#include <malloc.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <stdlib.h>
--- ./setosflags	Thu Nov 11 15:36:51 1999
+++ ../.build/macosx26_gcc31/shift/2003/setosflags	Wed Oct  8 13:33:36 2003
@@ -37,6 +37,9 @@
 	OSF1)	OSMAJNO=`echo $sysrel | awk '{ print substr($1,2,1)}'`
 		OSMINNO=`echo $sysrel | awk '{ print substr($1,4)}'`
 		;;
+	Darwin)	OSMAJNO=`echo $sysrel | awk '{ print substr($1,1,1)}'`
+		OSMINNO=`echo $sysrel | awk '{ print substr($1,3)}'`
+		;;
 	Linux)	OSMAJNO=`echo $sysrel | awk '{ print substr($1,1,1)}'`
 		OSMINNO=`echo $sysrel | awk '{ print substr($1,3)}'`
 		;;
--- ./tape/mounttape.c	Thu Nov 11 15:41:22 1999
+++ ../.build/macosx26_gcc31/shift/2003/tape/mounttape.c	Wed Oct  8 13:33:36 2003
@@ -615,7 +615,7 @@
 #if SONYRAW
 		if (sonyraw) break;
 #endif
-#if linux
+#if linux && !__APPLE__
 		mtop.mt_op = MTSETBLK;
 		mtop.mt_count = 0;
 		ioctl (tapefd, MTIOCTOP, &mtop);	/* set variable block size */
--- copytape/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ copytape/Imakefile	Mon Oct 20 22:44:55 2003
@@ -25,10 +25,10 @@
 #endif
 #else
 #if linux
-LIBS = -L../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../lib -lshift $(ADNSLIB) 
 #else
 #if SOLARIS
-LIBS = -L../lib -lshift -lsocket -lnsl
+LIBS = -L../lib -lshift -lsocket 
 #else
 #if defined(__alpha) && defined(__osf__)
 LIBS = -no_so -L../lib -lshift -so_archive
--- db/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ db/Imakefile	Mon Oct 20 22:44:55 2003
@@ -246,7 +246,7 @@
 #else
 #ifdef SOLARIS
 OSFLAG         = -DSOLARIS
-OSLIBS         = -lresolv -lsocket -lnsl
+OSLIBS         = -lresolv -lsocket 
 #else
 #ifdef _WIN32
 OSFLAG         = -D_WIN32
--- examples/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ examples/Imakefile	Mon Oct 20 22:44:55 2003
@@ -35,7 +35,7 @@
 LDLIBS = -L../lib -lshift -lsun $(ULTRA)
 #else /* ! hpux && ! cray && ! sgi */
 #if linux
-LIBS = -L../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../lib -lshift $(ADNSLIB) 
 #else
 #if SOLARIS
 LDLIBS = -L../lib -lshift -lsocket $(ULTRA)
--- hsmtools/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ hsmtools/Imakefile	Mon Oct 20 22:44:55 2003
@@ -22,13 +22,13 @@
 LIBS = -L../lib -lshift
 #else
 #if linux
-LIBS = -L../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../lib -lshift $(ADNSLIB) 
 #else
 #if sgi
 LIBS = -L../lib -lshift
 #else
 #if SOLARIS
-LIBS = -L../lib -lshift -lsocket -lnsl
+LIBS = -L../lib -lshift -lsocket 
 #else
 #if _WIN32
 LIBS = ..\lib\shift.lib wsock32.lib advapi32.lib
--- monitor/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ monitor/Imakefile	Mon Oct 20 22:44:55 2003
@@ -81,7 +81,7 @@
 #else
 #ifdef SOLARIS
 
-OSLIBS         = -lresolv -lsocket -lnsl
+OSLIBS         = -lresolv -lsocket 
 #else
 #ifdef _WIN32
 
--- msg/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ msg/Imakefile	Mon Oct 20 22:44:55 2003
@@ -16,7 +16,7 @@
 CFLAGS 		= -O -I$(INCLUDES) $(CCOPTFLAGS) $(DEFCFLAGS) $(MSGFLAGS) $(MTCCFLAGS)
 LDFLAGS     = $(MTLDFLAGS)
 #if SOLARIS
-LDLIBS		= -L../common -lcommon -lsocket -lnsl $(MTLDLIBS)
+LDLIBS		= -L../common -lcommon -lsocket  $(MTLDLIBS)
 #else
 #if linux
 LDLIBS		= -L../common -lcommon $(ADNSLIB) $(MTLDLIBS)
--- ns/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ ns/Imakefile	Mon Oct 20 22:44:55 2003
@@ -19,13 +19,13 @@
 LIBS = -L../lib -lshift
 #else
 #if linux
-LIBS = -L../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../lib -lshift $(ADNSLIB) 
 #else
 #if sgi
 LIBS = -L../lib -lshift
 #else
 #if SOLARIS
-LIBS = -L../lib -lshift -lsocket -lnsl
+LIBS = -L../lib -lshift -lsocket 
 #else
 #if _WIN32
 LIBS = ..\lib\shift.lib wsock32.lib advapi32.lib
--- rfio/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ rfio/Imakefile	Mon Oct 20 22:44:55 2003
@@ -105,16 +105,16 @@
 LDLIBS = -no_so -L../lib -lshift -so_archive
 #else
 #if linux
-LDLIBS = -L../lib -lshift $(ADNSLIB) -lnsl $(MTLDLIBS)
+LDLIBS = -L../lib -lshift $(ADNSLIB)  $(MTLDLIBS)
 #else
 #if SOLARIS
-LDLIBS = -L../lib -lshift -lsocket -lnsl
+LDLIBS = -L../lib -lshift -lsocket 
 #else
 #if sgi
 LDLIBS =  -L../lib -lshift $(MTLDLIBS)
 #else
 #if __Lynx__
-LDLIBS = -L../lib -lshift -lnsl -lbsd
+LDLIBS = -L../lib -lshift  -lbsd
 #else
 #if _WIN32
 LDLIBS = ..\lib\shift.lib ws2_32.lib advapi32.lib mpr.lib
--- rmc/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ rmc/Imakefile	Mon Oct 20 22:44:55 2003
@@ -8,10 +8,10 @@
 
 INCLUDES = FileName(..,h)
 #if linux
-LIBS = -L../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../lib -lshift $(ADNSLIB) 
 #else
 #if SOLARIS
-LIBS = -L../lib -lshift -lsocket -lnsl
+LIBS = -L../lib -lshift -lsocket 
 #else
 #if defined(_AIX) && defined(_IBMR2)
 LIBS = -L../lib -lshift -lodm
--- rtcopy/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ rtcopy/Imakefile	Mon Oct 20 22:44:55 2003
@@ -106,7 +106,7 @@
 THREADFLAGS = $(MTCCFLAGS)
 
 #if SOLARIS
-LDLIBS = -lsocket -lnsl
+LDLIBS = -lsocket 
 #else
 #if _AIX
 #if defined(_IBMR2)
@@ -114,7 +114,7 @@
 #endif
 #else
 #if linux
-LDLIBS = $(ADNSLIB) -lnsl
+LDLIBS = $(ADNSLIB) 
 #else
 #if _WIN32
 LDLIBS = wsock32.lib ws2_32.lib advapi32.lib
--- stage/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ stage/Imakefile	Mon Oct 20 22:44:55 2003
@@ -35,13 +35,13 @@
 #endif
 #else
 #if linux
-OSLIBS = DepLibraryTargetName(lib,shift) $(ADNSLIB) -lnsl
+OSLIBS = DepLibraryTargetName(lib,shift) $(ADNSLIB) 
 #else
 #if SOLARIS
-OSLIBS = DepLibraryTargetName(lib,shift) -lsocket -lnsl
+OSLIBS = DepLibraryTargetName(lib,shift) -lsocket 
 #else
 #if __Lynx__
-OSLIBS = DepLibraryTargetName(lib,shift) -lnsl -lbsd
+OSLIBS = DepLibraryTargetName(lib,shift)  -lbsd
 #else
 #if _WIN32
 OSLIBS = DepLibraryTargetName(lib,shift) ws2_32.lib wsock32.lib advapi32.lib mpr.lib
--- sysreq/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ sysreq/Imakefile	Mon Oct 20 22:44:55 2003
@@ -21,13 +21,13 @@
 #endif
 #else
 #if linux
-LDLIBS = -L../common -lcommon -lnsl $(ADNSLIB) $(MTLDLIBS)
+LDLIBS = -L../common -lcommon  $(ADNSLIB) $(MTLDLIBS)
 #else
 #if SOLARIS
-LDLIBS = -L../common -lcommon -lsocket -lnsl $(MTLDLIBS)
+LDLIBS = -L../common -lcommon -lsocket  $(MTLDLIBS)
 #else
 #if __Lynx__
-LDLIBS = -L../common -lcommon -lnsl -lbsd $(MTLDLIBS)
+LDLIBS = -L../common -lcommon  -lbsd $(MTLDLIBS)
 #else
 #if _WIN32
 LDLIBS = ..\common\common.lib wsock32.lib advapi32.lib mpr.lib $(MTLDLIBS)
--- tape/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ tape/Imakefile	Mon Oct 20 22:44:55 2003
@@ -24,10 +24,10 @@
 VDQMFLG = -DVDQM
 #endif
 #if linux
-LIBS = -L../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../lib -lshift $(ADNSLIB) 
 #else
 #if SOLARIS
-LIBS = -L../lib -lshift -lsocket -lnsl
+LIBS = -L../lib -lshift -lsocket 
 #else
 #if defined(_AIX) && defined(_IBMR2)
 LIBS = -L../lib -lshift -lodm
--- test/common/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ test/common/Imakefile	Mon Oct 20 22:44:55 2003
@@ -34,7 +34,7 @@
 #else
 #ifdef SOLARIS
 COMM Because of h_errno external and POSIX4 library
-SLIB         = -lnsl -lposix4
+SLIB         =  -lposix4
 #else
 #ifdef __osf__
 SLIB         = -lrt
--- test/db/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ test/db/Imakefile	Mon Oct 20 22:44:55 2003
@@ -49,7 +49,7 @@
 #else
 #ifdef SOLARIS
 OSFLAG         = -DSOLARIS
-OSLIBS         = -lresolv -lsocket -lnsl
+OSLIBS         = -lresolv -lsocket 
 #else
 #ifdef _WIN32
 OSFLAG         = -D_WIN32
--- test/ns/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ test/ns/Imakefile	Mon Oct 20 22:44:55 2003
@@ -8,10 +8,10 @@
 
 INCLUDES = FileName(FileName(..,..),h)
 #if linux
-LIBS = -L../../ns -lns -L../../common -lcommon $(ADNSLIB) -lnsl
+LIBS = -L../../ns -lns -L../../common -lcommon $(ADNSLIB) 
 #else
 #if SOLARIS
-LIBS = -L../../ns -lns -L../../common -lcommon -lsocket -lnsl
+LIBS = -L../../ns -lns -L../../common -lcommon -lsocket 
 #else
 #if _WIN32
 LIBS = ..\..\ns\ns.lib ..\..\common\common.lib wsock32.lib advapi32.lib
--- test/rfio/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ test/rfio/Imakefile	Mon Oct 20 22:44:55 2003
@@ -8,10 +8,10 @@
 
 INCLUDES = FileName(FileName(..,..),h)
 #if linux
-LIBS = -L../../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../../lib -lshift $(ADNSLIB) 
 #else
 #if SOLARIS
-LIBS = -L../../lib -lshift -lsocket -lnsl
+LIBS = -L../../lib -lshift -lsocket 
 #else
 #if defined(__alpha) && defined(__osf__)
 LIBS = -no_so -L../../lib -lshift -so_archive
--- test/stage/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ test/stage/Imakefile	Mon Oct 20 22:44:55 2003
@@ -16,13 +16,13 @@
 #endif
 #else
 #if linux
-LIBS = DepLibraryTargetName(FileName(..,lib),shift) $(ADNSLIB) -lnsl
+LIBS = DepLibraryTargetName(FileName(..,lib),shift) $(ADNSLIB) 
 #else
 #if SOLARIS
-LIBS = DepLibraryTargetName(FileName(..,lib),shift) -lsocket -lnsl
+LIBS = DepLibraryTargetName(FileName(..,lib),shift) -lsocket 
 #else
 #if __Lynx__
-LIBS = DepLibraryTargetName(FileName(..,lib),shift) -lnsl -lbsd
+LIBS = DepLibraryTargetName(FileName(..,lib),shift)  -lbsd
 #else
 #if _WIN32
 LIBS = DepLibraryTargetName(FileName(..,lib),shift) wsock32.lib advapi32.lib mpr.lib
--- test/tape/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ test/tape/Imakefile	Mon Oct 20 22:44:55 2003
@@ -8,10 +8,10 @@
 
 INCLUDES = FileName(FileName(..,..),h)
 #if linux
-LIBS = -L../../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../../lib -lshift $(ADNSLIB) 
 #else
 #if SOLARIS
-LIBS = -L../../lib -lshift -lsocket -lnsl
+LIBS = -L../../lib -lshift -lsocket 
 #else
 #if defined(_AIX) && defined(_IBMR2)
 LIBS = -L../../lib -lshift -lodm
--- test/vmgr/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ test/vmgr/Imakefile	Mon Oct 20 22:44:55 2003
@@ -9,10 +9,10 @@
 INCLUDES = FileName(FileName(..,..),h)
 
 #if linux
-LIBS = -L../../vmgr -lvmgr -L../../common -lcommon $(ADNSLIB) -lnsl
+LIBS = -L../../vmgr -lvmgr -L../../common -lcommon $(ADNSLIB) 
 #else
 #if SOLARIS
-LIBS = -L../../vmgr -lvmgr -L../../common -lcommon -lsocket -lnsl
+LIBS = -L../../vmgr -lvmgr -L../../common -lcommon -lsocket 
 #else
 #if _WIN32
 LIBS = ..\..\vmgr\vmgr.lib ..\..\common\common.lib wsock32.lib advapi32.lib
--- upv/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ upv/Imakefile	Mon Oct 20 22:44:55 2003
@@ -17,13 +17,13 @@
 LIBS = -L../lib -lshift
 #else
 #if linux
-LIBS = -L../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../lib -lshift $(ADNSLIB) 
 #else
 #if sgi
 LIBS = -L../lib -lshift
 #else
 #if SOLARIS
-LIBS = -L../lib -lshift -lsocket -lnsl
+LIBS = -L../lib -lshift -lsocket 
 #else
 #if _WIN32
 LIBS = ..\lib\shift.lib wsock32.lib advapi32.lib
--- vdqm/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ vdqm/Imakefile	Mon Oct 20 22:44:55 2003
@@ -27,10 +27,10 @@
 THREADFLAGS = $(MTCCFLAGS)
 
 #if SOLARIS
-LDLIBS = -lsocket -lnsl
+LDLIBS = -lsocket 
 #else
 #if linux
-LDLIBS = $(ADNSLIB) -lnsl
+LDLIBS = $(ADNSLIB) 
 #else
 #if _WIN32
 LDLIBS = ws2_32.lib advapi32.lib mpr.lib
--- vmgr/Imakefile.orig	Mon Oct 20 22:44:30 2003
+++ vmgr/Imakefile	Mon Oct 20 22:44:55 2003
@@ -17,13 +17,13 @@
 LIBS = -L../lib -lshift
 #else
 #if linux
-LIBS = -L../lib -lshift $(ADNSLIB) -lnsl
+LIBS = -L../lib -lshift $(ADNSLIB) 
 #else
 #if sgi
 LIBS = -L../lib -lshift
 #else
 #if SOLARIS
-LIBS = -L../lib -lshift -lsocket -lnsl
+LIBS = -L../lib -lshift -lsocket 
 #else
 #if _WIN32
 LIBS = ..\lib\shift.lib wsock32.lib advapi32.lib
--- common/Imakefile.orig	Mon Oct 20 22:51:27 2003
+++ common/Imakefile	Mon Oct 20 22:52:15 2003
@@ -147,8 +147,8 @@
 		echo $(CC) $(CFLAGS) -DSCSIINC="\"/usr/src/linux/include/scsi/sg.h\"" -c sendscsicmd.c ;\
 		$(CC) $(CFLAGS) -DSCSIINC="\"/usr/src/linux/include/scsi/sg.h\"" -c sendscsicmd.c ;\
 	else \
-		echo $(CC) $(CFLAGS) -DSCSIINC="\"/usr/include/scsi/sg.h\"" -c sendscsicmd.c ;\
-		$(CC) $(CFLAGS) -DSCSIINC="\"/usr/include/scsi/sg.h\"" -c sendscsicmd.c ;\
+		echo $(CC) $(CPPFLAGS) $(CFLAGS) -DSCSIINC="\"/usr/include/scsi/sg.h\"" -c sendscsicmd.c ;\
+		$(CC) $(CPPFLAGS) $(CFLAGS) -DSCSIINC="\"/usr/include/scsi/sg.h\"" -c sendscsicmd.c ;\
 	fi
 #endif
 #ifndef _WIN32
@@ -157,15 +157,15 @@
 COMM explicitely the debug option
 COMM
 Cthread.Osuf: Cthread.c
-	$(CC) $(CFLAGS) -g $(LINUXOPTIM) -c Cthread.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -g $(LINUXOPTIM) -c Cthread.c
 Cpool.Osuf: Cpool.c
-	$(CC) $(CFLAGS) -g $(LINUXOPTIM) -c Cpool.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -g $(LINUXOPTIM) -c Cpool.c
 Csched.Osuf: Csched.c
-	$(CC) $(CFLAGS) -g $(LINUXOPTIM) -c Csched.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -g $(LINUXOPTIM) -c Csched.c
 Cmutex.Osuf: Cmutex.c
-	$(CC) $(CFLAGS) -g $(LINUXOPTIM) -c Cmutex.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -g $(LINUXOPTIM) -c Cmutex.c
 Csetprocname.Osuf: Csetprocname.c
-	$(CC) $(CFLAGS) -g $(LINUXOPTIM) -c Csetprocname.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -g $(LINUXOPTIM) -c Csetprocname.c
 #endif
 #if _WIN32
 Cinitservice.c:	..\win32\Cinitservice.c
--- rfio/Imakefile.orig	Mon Oct 20 22:51:27 2003
+++ rfio/Imakefile	Mon Oct 20 22:52:21 2003
@@ -349,7 +349,7 @@
 #if ((defined(__osf__) && defined(__alpha) && (OSMajorVersion == 4)) || defined(linux))
 #if RFIODaemonRealTime
 rfio_calls.Osuf: rfio_calls.c
-	$(CC) $(CFLAGS) -DRFIODaemonRealTime -c rfio_calls.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -DRFIODaemonRealTime -c rfio_calls.c
 #endif
 #endif
 #if _WIN32
--- rtcopy/Imakefile.orig	Mon Oct 20 22:51:27 2003
+++ rtcopy/Imakefile	Mon Oct 20 22:52:31 2003
@@ -250,7 +250,7 @@
 	$(CC) -c $(CPPFLAGS) $(CCOPTIM) $(THREADFLAGS) -I$(INCLUDES) dumptape.c
 #if BuildRtstat
 rtstat.Osuf:		rtstat.c ../h/vdqm.h ../h/osdep.h ../h/net.h
-	$(CC) -c $(CFLAGS) rtstat.c
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) rtstat.c
 #endif
 
 NormalProgramTarget(tpread,$(TPREAD_OBJS) $(RTCPLIB_OBJS),$(GLOBLIB),$(MTLDFLAGS) $(GLOBLIB) $(LDLIBS) $(MTLDLIBS))
--- stage/Imakefile.orig	Mon Oct 20 22:51:27 2003
+++ stage/Imakefile	Mon Oct 20 22:52:50 2003
@@ -551,7 +551,7 @@
 #if UseStageCdb
 STGCONVERT_DB_OBJS = FileName(FileName(..,db),Cdb_hash.Osuf) FileName(FileName(..,db),Cdb_error.Osuf) FileName(FileName(..,db),Cdb_login.Osuf) FileName(FileName(..,db),Cdb_lock.Osuf) FileName(FileName(..,db),Cdb_util.Osuf) FileName(FileName(..,db),Cdb_ana.Osuf) FileName(FileName(..,db),Cdb_db.Osuf) FileName(FileName(..,db),Cdb_wrapio.Osuf)
 stgconvert.Osuf: stgconvert.c
-	$(CC) -o stgconvert.Osuf $(CFLAGS) -DSTGCONVERT -c stgconvert.c
+	$(CC) -o stgconvert.Osuf $(CFLAGS) $(CPPFLAGS) -DSTGCONVERT -c stgconvert.c
 STGCONVERT_OBJS = stgconvert.Osuf $(DBOBJS) $(STGCONVERT_DB_OBJS) $(GLOBLIB)
 #if defined(_WIN32)
 $(STGCONVERT_DB_OBJS): FORCE
@@ -571,7 +571,7 @@
 NormalProgramTarget(stgdump,stgdump.Osuf $(DBOBJS_STGD) stglogit_term.Osuf,$(GLOBLIB),$(LIBS))
 #if hpux
 check_Cdbentry.Osuf:
-	$(CC) -o check_Cdbentry.Osuf -Ae $(CFLAGS) -c check_Cdbentry.c
+	$(CC) -o check_Cdbentry.Osuf -Ae $(CFLAGS) $(CPPFLAGS) -c check_Cdbentry.c
 #endif
 #endif
 
--- tape/Imakefile.orig	Mon Oct 20 22:51:27 2003
+++ tape/Imakefile	Mon Oct 20 22:53:29 2003
@@ -316,34 +316,34 @@
 sendscsicmd.Osuf:	sendscsicmd.c ../h/Ctape.h ../h/scsictl.h
 	@if [ -f "/usr/src/linux/include/scsi/sg.h" ] ;\
 	then \
-		echo $(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DTAPE -DSCSIINC="\"/usr/src/linux/include/scsi/sg.h\"" sendscsicmd.c ;\
-		$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DTAPE -DSCSIINC="\"/usr/src/linux/include/scsi/sg.h\"" sendscsicmd.c ;\
+		echo $(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DTAPE -DSCSIINC="\"/usr/src/linux/include/scsi/sg.h\"" sendscsicmd.c ;\
+		$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DTAPE -DSCSIINC="\"/usr/src/linux/include/scsi/sg.h\"" sendscsicmd.c ;\
 	else \
-		echo $(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DTAPE -DSCSIINC="\"/usr/include/scsi/sg.h\"" sendscsicmd.c ;\
-		$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DTAPE -DSCSIINC="\"/usr/include/scsi/sg.h\"" sendscsicmd.c ;\
+		echo $(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DTAPE -DSCSIINC="\"/usr/include/scsi/sg.h\"" sendscsicmd.c ;\
+		$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DTAPE -DSCSIINC="\"/usr/include/scsi/sg.h\"" sendscsicmd.c ;\
 	fi
 #else
 sendscsicmd.Osuf:	sendscsicmd.c ../h/Ctape.h ../h/scsictl.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DTAPE sendscsicmd.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DTAPE sendscsicmd.c
 #endif
 usrreadlbl.Osuf:	usrreadlbl.c ../h/Ctape.h ../h/serrno.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE $(PCTAFLAG) $(ADSTARFLG) usrreadlbl.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE $(PCTAFLAG) $(ADSTARFLG) usrreadlbl.c
 usrrwndtape.Osuf:	usrrwndtape.c ../h/Ctape.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrrwndtape.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrrwndtape.c
 usrskiptape.Osuf:	usrskiptape.c ../h/Ctape.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrskiptape.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrskiptape.c
 usrsmcsubr.Osuf:	usrsmcsubr.c ../h/Ctape.h ../h/scsictl.h ../h/smc.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrsmcsubr.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrsmcsubr.c
 usrsmcsubr2.Osuf:	usrsmcsubr2.c ../h/Ctape.h ../h/smc.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrsmcsubr2.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrsmcsubr2.c
 usrtperror.Osuf:	usrtperror.c ../h/Ctape.h ../h/serrno.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrtperror.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrtperror.c
 usrusrmsg.Osuf:	usrusrmsg.c ../h/Ctape.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrusrmsg.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrusrmsg.c
 usrwritelbl.Osuf:	usrwritelbl.c ../h/Ctape.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrwritelbl.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrwritelbl.c
 usrwrttpmrk.Osuf:	usrwrttpmrk.c ../h/Ctape.h
-	$(CC) -c -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrwrttpmrk.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(MTCCFLAGS) -DNOTRACE usrwrttpmrk.c
 
 LinkFile(ProgramTargetName(dumptape),ProgramTargetName(tpdump))
 
--- vdqm/Imakefile.orig	Mon Oct 20 22:51:27 2003
+++ vdqm/Imakefile	Mon Oct 20 22:53:37 2003
@@ -190,7 +190,7 @@
 	$(CP) vdqm_SendRecv.c vdqmc_SendRecv.c
 
 vdqmc_SendRecv.Osuf: vdqmc_SendRecv.c 
-	$(CC) -c -I$(INCLUDES) $(THREADFLAGS) vdqmc_SendRecv.c
+	$(CC) -c $(CPPFLAGS) -I$(INCLUDES) $(THREADFLAGS) vdqmc_SendRecv.c
 #endif
 
 COMM DO NOT DELETE THIS LINE -- make  depend  depends  on  it.
--- upv/Cupvlist.c.orig	Mon Oct 20 23:16:39 2003
+++ upv/Cupvlist.c	Mon Oct 20 23:16:46 2003
@@ -10,12 +10,12 @@
 /*      Cupvlist - list privilege entries */
 
 #include <errno.h> 
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
--- msg/oper.c.orig	Mon Oct 20 23:17:42 2003
+++ msg/oper.c	Mon Oct 20 23:18:01 2003
@@ -80,7 +80,7 @@
 #if defined(HPUX10)
 #undef CR0
 #endif
-#include <termio.h>
+#include <termios.h>
 static struct termio termargs  ;	/* Terminal parameters		*/
 #define ERASE termargs.c_cc[VERASE] 	/* Current erase character	*/
 #define KILL  termargs.c_cc[VKILL]  	/* Current kill character	*/
--- common/Cinitdaemon.c.orig	Mon Oct 20 23:19:05 2003
+++ common/Cinitdaemon.c	Mon Oct 20 23:19:50 2003
@@ -44,7 +44,7 @@
 #if hpux
         c = setpgrp3();
 #else
-        c = setpgrp();
+        c = setpgrp(0, getpid ());
 #endif
 #endif
         for (c = 0; c < maxfds; c++)
--- copytape/ctpdaemon.c.orig	Mon Oct 20 23:19:05 2003
+++ copytape/ctpdaemon.c	Mon Oct 20 23:20:06 2003
@@ -106,7 +106,7 @@
 		exit (1);
 	} else
 		if (c > 0) exit (0);
-#if (defined(sun) && !defined(SOLARIS)) || defined(ultrix) || defined(_IBMESA)
+#if (defined(sun) && !defined(SOLARIS)) || defined(ultrix) || defined(_IBMESA) || defined (__APPLE__)
 	c = setpgrp(0, getpid());
 #else
 #if (defined(__osf__) && defined(__alpha)) || defined(linux)
--- rtcopy/rtcpd_MainCntl.c.orig	Mon Oct 20 23:19:05 2003
+++ rtcopy/rtcpd_MainCntl.c	Mon Oct 20 23:20:29 2003
@@ -1420,7 +1420,7 @@
     extern int client_magic;
 
 #if !defined(_WIN32)
-    (void)setpgrp();
+    (void)setpgrp(0, getpid ());
 #endif /* _WIN32 */
     AbortFlag = 0;
     rtcp_log(LOG_DEBUG,"rtcpd_MainCntl() called\n");
--- tape/getdrvstatus.c.orig	Mon Oct 20 23:23:15 2003
+++ tape/getdrvstatus.c	Mon Oct 20 23:23:23 2003
@@ -94,7 +94,7 @@
 		return (WRITE_ENABLE);
 }
 #else
-#if sun
+#if sun || __APPLE__
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/mtio.h>
--- vmgr/vmgrenterlibrary.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrenterlibrary.c	Mon Oct 20 23:26:53 2003
@@ -8,10 +8,10 @@
 #endif /* not lint */
 
 /*	vmgrenterlibrary - define a new tape library */
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
-#include <sys/types.h>
 #include "Cgetopt.h"
 #include "serrno.h"
 #include "vmgr_api.h"
--- vmgr/vmgrenterpool.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrenterpool.c	Mon Oct 20 23:26:59 2003
@@ -8,10 +8,10 @@
 #endif /* not lint */
 
 /*	vmgrenterpool - define a new tape pool */
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
-#include <sys/types.h>
 #include "Cgetopt.h"
 #include "serrno.h"
 #include "vmgr_api.h"
--- vmgr/vmgrlistdenmap.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrlistdenmap.c	Mon Oct 20 23:27:06 2003
@@ -9,10 +9,10 @@
 
 /*      vmgrlistdenmap - list triplets model/media_letter/density */
 #include <errno.h>
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
--- vmgr/vmgrlistdgnmap.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrlistdgnmap.c	Mon Oct 20 23:27:11 2003
@@ -9,10 +9,10 @@
 
 /*      vmgrlistdgnmap - list triplets dgn/model/library */
 #include <errno.h>
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
--- vmgr/vmgrlistlibrary.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrlistlibrary.c	Mon Oct 20 23:27:15 2003
@@ -9,11 +9,11 @@
 
 /*	vmgrlistlibrary - query a given library or list all existing tape libraries */
 #include <errno.h>
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
--- vmgr/vmgrlistmodel.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrlistmodel.c	Mon Oct 20 23:27:19 2003
@@ -9,11 +9,11 @@
 
 /*      vmgrlistmodel - list cartridge model entries */
 #include <errno.h>
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
--- vmgr/vmgrlistpool.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrlistpool.c	Mon Oct 20 23:27:24 2003
@@ -9,11 +9,11 @@
 
 /*	vmgrlistpool - query a given pool or list all existing tape pools */
 #include <errno.h>
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
--- vmgr/vmgrmodifylibrary.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrmodifylibrary.c	Mon Oct 20 23:27:33 2003
@@ -8,10 +8,10 @@
 #endif /* not lint */
 
 /*	vmgrmodifylibrary - modify an existing tape library definition */
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
-#include <sys/types.h>
 #include "Cgetopt.h"
 #include "serrno.h"
 #include "vmgr_api.h"
--- vmgr/vmgrmodifypool.c.orig	Mon Oct 20 23:24:49 2003
+++ vmgr/vmgrmodifypool.c	Mon Oct 20 23:27:38 2003
@@ -8,10 +8,10 @@
 #endif /* not lint */
 
 /*	vmgrmodifypool - modify an existing tape pool definition */
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
-#include <sys/types.h>
 #include "Cgetopt.h"
 #include "serrno.h"
 #include "vmgr_api.h"
 
--- common/Cuuid.c.orig	Sun Nov 16 16:16:58 2003
+++ common/Cuuid.c	Sun Nov 16 16:13:25 2003
@@ -33,7 +33,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
-#ifndef _WIN32
+#if !defined _WIN32 && !defined __APPLE__
 #include <sys/sysinfo.h>
 #endif
 #if (defined(aix) || defined(hpux))
@@ -766,7 +766,9 @@
 {
     MD5_CTX c;
     typedef struct {
+#ifndef __APPLE__
         struct sysinfo s;
+#endif
         struct timeval t;
         char hostname[257];
     } randomness;
@@ -796,7 +798,9 @@
 	}
 #endif
 #else /* aix || hpux */
+#ifndef __APPLE__
     sysinfo(&r.s);
+#endif
 #endif
     gettimeofday(&r.t, (struct timezone *)0);
     gethostname(r.hostname, 256);
--- common/socket_timeout.c.orig	Sun Nov 16 16:17:32 2003
+++ common/socket_timeout.c	Sun Nov 16 16:18:10 2003
@@ -7,7 +7,7 @@
 static char sccsid[] = "@(#)socket_timeout.c,v 1.22 2003/10/29 13:06:34 CERN IT-PDP/DM Jean-Damien Durand";
 #endif /* not lint */
 
-#if defined(linux)
+#if defined(linux) && !defined (__APPLE__)
 #define USE_POLL_INSTEAD_OF_SELECT
 /* #define __FD_SETSIZE 65536 */
 #endif
--- rfio/readdir.c.orig	Sun Nov 16 16:19:19 2003
+++ rfio/readdir.c	Sun Nov 16 16:20:08 2003
@@ -151,7 +151,7 @@
    return(de);
 }
 
-#if !defined(SOLARIS) && !defined(linux)
+#if !defined(SOLARIS) && (!defined(linux) || defined (__APPLE__))
 struct dirent DLL_DECL *rfio_readdir64(dirp)
 RDIR *dirp;
 {
--- vmgr/vmgrdeltag.c.orig	Sun Nov 16 16:31:29 2003
+++ vmgr/vmgrdeltag.c	Sun Nov 16 16:31:45 2003
@@ -8,12 +8,12 @@
 #endif /* not lint */
 
 /*	vmgrdeltag - delete a tag associated with a tape volume */
+#include <sys/types.h>
 #include <errno.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
--- vmgr/vmgrsettag.c.orig	Sun Nov 16 16:31:29 2003
+++ vmgr/vmgrsettag.c	Sun Nov 16 16:31:50 2003
@@ -8,12 +8,12 @@
 #endif /* not lint */
 
 /*	vmgrsettag - add/replace a tag associated with a tape volume */
+#include <sys/types.h>
 #include <errno.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
--- vmgr/vmgrgettag.c.orig	Sun Nov 16 16:31:29 2003
+++ vmgr/vmgrgettag.c	Sun Nov 16 16:31:54 2003
@@ -8,12 +8,12 @@
 #endif /* not lint */
 
 /*	vmgrgettag - get the tag associated with a tape volume */
+#include <sys/types.h>
 #include <errno.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/types.h>
 #if defined(_WIN32)
 #include <winsock2.h>
 #endif
