### PATCH uuid-osx
--- lib/uuid/Makefile.in        Mon Aug 18 16:06:26 2003
+++ lib/uuid/Makefile.in        Mon Aug 18 16:08:25 2003
@@ -73,6 +73,13 @@
 @MAKEFILE_PROFILE@
 @MAKEFILE_CHECKER@
 
+all:: $(LIBRARY).dylib
+install:: all
+	$(INSTALL_DATA) libuuid.dylib $(DESTDIR)$(libdir)/libuuid.dylib
+	$(CHMOD) 755 $(DESTDIR)$(libdir)/libuuid.dylib
+$(LIBRARY).dylib: $(OBJS)
+	$(CC) -dynamiclib -o $@ $(OBJS)
+
 .c.o:
 	$(CC) $(ALL_CFLAGS) -c $< -o $@
 @PROFILE_CMT@	$(CC) $(ALL_CFLAGS) -g -pg -o profiled/$*.o -c $<

